Gems/WhiteBox/Code/Source/Rendering/Atom/WhiteBoxAtomRenderMesh.cpp

```diff
@@ -261,13 +262,37 @@ namespace WhiteBox
         m_meshFeatureProcessor->SetTransform(m_meshHandle, worldFromLocal);
     }
 
+
+
     void AtomRenderMesh::UpdateMaterial(const WhiteBoxMaterial& material)
     {
+        if (m_meshFeatureProcessor)
+        {
+            auto& materialAssignment = m_materialMap[AZ::Render::DefaultMaterialAssignmentId];
+            materialAssignment.m_propertyOverrides[AZ::Name("baseColor.color")] = AZ::Color(material.m_tint);
+            materialAssignment.m_propertyOverrides[AZ::Name("baseColor.useTexture")] = material.m_useTexture;
+            if (materialAssignment.ApplyProperties())
```

Comment:
suggestion: A guide comment here might be useful to let the reader know what is happening, something like... `if ApplyProperties fails, defer updating the material assignment map to the next tick, and try applying properties again`